### PR TITLE
util units: simplify round_down_significant and improve docstring

### PR DESCRIPTION
**Pull request by pieleric**
From code golf session (only kepts the non-crazy part)

### CODE REVIEW

File: src/odemis/util/units.py [hunk 1]
```diff
@@ -66,20 +66,19 @@ def round_significant(x, n):
...
         return 0
 
     exp = n - math.ceil(math.log10(abs(x)))
-    if x > 0:
-        ret = math.floor(x * 10 ** exp) / (10 ** exp)
-    else:
-        ret = math.ceil(x * 10 ** exp) / (10 ** exp)
-    # assert(abs(ret) <= abs(x))
-    return ret
+    return math.trunc(x * 10 ** exp) / (10 ** exp)
```
Comment (BasHolweg) [inline src/odemis/util/units.py:81]:
```
>>> n = 2
>>> x =136
>>> exp = n - math.ceil(math.log10(abs(x)))
>>> math.trunc(x * 10 ** exp) / (10 ** exp)
130.0
```
For n = 2, x = 136. The resulting value is a float with more significant numbers then x. Is this expected behaviour. If yes could this be commented in the docstring. If not perhaps make it an interger if first significant decimal number is zero.

**Review by BasHolweg — CHANGES_REQUESTED** [review]

File: src/odemis/util/units.py (hunk 1 above)
Comment (pieleric) [reply src/odemis/util/units.py:81]: That's per design. It *always* returns a float. Actually, in Python you can almost always abstract float vs integer, so we don't really care. The default print() of Python show them differently, but the Odemis GUI doesn't use the default print function ;-) Here, the final number has only two significant figures: 1 & 3. zeros are used to replace non-significant numbers.

Comment (tepals) [reply src/odemis/util/units.py:81]: Could you add an example like this to the docstring? It's also not completely clear to me, just by reading the docstring.

Comment (pieleric) [reply src/odemis/util/units.py:84]: Added 2 examples to the docstring.

**Review by tepals — APPROVED** [review]

**Pull request merged**